services/motion/builtin/builtin.go

```diff
@@ -231,24 +236,111 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
-	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
-	}
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// TODO: if we disentangle creation of the kinematic base from the plan call we make we don't need this and can do this once
+	var kinematicBase kinematicbase.KinematicBase = nil
+
+	// Shared variables for planning and execution threads, are all guarded with a mutex
+	var planMu sync.Mutex
+	var moveCtx context.Context
+	var currentPlan [][]referenceframe.Input = nil
+
+	// Shared variable between threads that can trigger replanning and planning thread
+	var replanFlag atomic.Bool
+	replanFlag.Store(true)
+
+	var successFlag atomic.Bool
+	successFlag.Store(false)
+	errChan := make(chan error)
+
+	ms.backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+
+		// maybe move this into the restart tickers function
+		cancelCtx, cancelFn := context.WithCancel(context.Background())
+		ms.cancelFn = cancelFn
```

Comment:
In the instance where a subsequent MoveOnGlobe call is received, this will not interact well, allowing a zombie unkillable background process.

At a minimum this should call the existing `ms.cancelFn` if it is non-nil before swapping it out.

Reply:
Thats a good catch, I'll add that.

Reply:
Note that this does, for the first time, enforce that only a single move call may be active at one time.